src/UnitfulRecipes.jl

```diff
@@ -50,11 +50,28 @@ const AVec = AbstractVector
 const AMat{T} = AbstractArray{T,2} where T
 @recipe function f(x::AVec, y::AVec, z::AMat{T}) where T <: Quantity
     u = get(plotattributes, :zunit, unit(eltype(z)))
+    ustripattribute!(plotattributes, :clims, u)
     z = fixaxis!(plotattributes, z, :z)
     append_unit_if_needed!(plotattributes, :colorbar_title, u)
     x, y, z
 end
 
+# Recipe for contour|heatmap|surface(z)
+# Caution: also gets use for plot(z::Matrix)
+@recipe function f(z::AMat{T}) where T <: Quantity
+    u = get(plotattributes, :zunit, unit(eltype(z)))
+    clims_types = (:contour, :contourf, :heatmap, :surface)
+    if get(plotattributes, :seriestype, :nothing) ∈ clims_types
+        ustripattribute!(plotattributes, :clims, u)
+        z = fixaxis!(plotattributes, z, :z)
+        append_unit_if_needed!(plotattributes, :colorbar_title, u)
+    else # plot(z::Matrix)
+        z = fixaxis!(plotattributes, z, :y)
+        append_unit_if_needed!(plotattributes, :y, u)
+    end
+    z
+end
```

Comment:
I think this is partially redundant with the recipe lines 11–14. Could you try and remove these lines here (59–73) and replace lines 11–14 with something like:

```julia
@recipe function f(::Type{T}, x::T) where T <: AbstractArray{<:Union{Missing,<:Quantity}}
    axisletter = plotattributes[:letter]   # x, y, or z
    if (axisletter == :z) && (get(plotattributes, :seriestype, :nothing) ∈ (:contour, :contourf, :heatmap, :surface))
        u = get(plotattributes, :zunit, unit(eltype(x)))
        ustripattribute!(plotattributes, :clims, u)
        append_unit_if_needed!(plotattributes, :colorbar_title, u)
    end
    fixaxis!(plotattributes, x, axisletter)
end
```

and see if this works the same?

Reply:
BTW I feel like this could actually be put in the `fixmarker` function (and maybe rename it)?

Reply:
You were correct that putting it in lines 11-14 was sufficient so I've done it that way now and cut the redundant code.

I didn't understand the suggestion about `fixmarker`, and I don't really understand recipes but I like to try to make PR when I report an issue just to try to be help in part.  I made this PR "editable by maintainers" so feel free to modify if you think there is an even simpler way.